junit-vintage-engine/src/main/java/org/junit/vintage/engine/discovery/IsPotentialJUnit4TestClass.java

```diff
@@ -22,12 +23,20 @@
 
 	@Override
 	public boolean test(Class<?> candidate) {
-		// Do not collapse into single return.
 		if (isAbstract(candidate))
 			return false;
 		if (!isPublic(candidate))
 			return false;
-		return !candidate.isMemberClass();
+		// Do not collapse into single return.
```

Comment:
I moved it here because this is where intellij generates the warning / quick fix for collapsing it. But if you want it at the top i'll move it.